<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from human_hand.xacro               | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="human_hand" xmlns:xacro="http://ros.org/wiki/xacro">
  <!--link name="world"/-->
  <link name="hand_link" type="fixed"/>
  <joint name="ahand_base_joint" type="fixed">
    <parent link="hand_link"/>
    <child link="ahand_palm_link"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
  </joint>
  <link name="ahand_palm_link">
    <visual>
      <origin rpy="0 0 0" xyz="0.04 -0.01 0"/>
      <material name="all">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <box size="0.08 0.07 .02"/>
      </geometry>
    </visual>
  </link>
  <joint name="index_dof0_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0.0934 0.02957  0"/>
    <parent link="ahand_palm_link"/>
    <child link="index_dof0_link"/>
    <limit effort="100" lower="-0.13963" upper="0.05236" velocity="100"/>
    <axis xyz="0 0 1"/>
  </joint>
  <link name="index_dof0_link">
    <visual>
      <origin rpy="0.0 0.0 0.0 " xyz="0 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <sphere radius="0.004" xyz="0 0 0"/>
      </geometry>
    </visual>
  </link>
  <joint name="index_dof1_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="index_dof0_link"/>
    <child link="index_dof1_link"/>
    <limit effort="100" lower="-0.13963" upper="0.05236" velocity="100"/>
    <axis xyz="0 -1 0"/>
  </joint>
  <link name="index_dof1_link">
    <visual>
      <origin rpy="0.0 0.0 0.0 " xyz="0 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <sphere radius="0.004" xyz="0 0 0"/>
      </geometry>
    </visual>
    <visual>
      <origin rpy="0.0 1.57079632679 0.0 " xyz="0.02397 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <cylinder length="0.04794" radius="0.002"/>
      </geometry>
    </visual>
  </link>
  <joint name="index_dof2_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0.04794 0 0"/>
    <parent link="index_dof1_link"/>
    <child link="index_dof2_link"/>
    <limit effort="100" lower="-1.5708" upper="0.0" velocity="100"/>
    <axis xyz="0 -1 0"/>
  </joint>
  <link name="index_dof2_link">
    <visual>
      <origin rpy="0.0 0.0 0.0 " xyz="0 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <sphere radius="0.004" xyz="0 0 0"/>
      </geometry>
    </visual>
    <visual>
      <origin rpy="0.0 1.57079632679 0.0 " xyz="0.014135 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <cylinder length="0.02827" radius="0.002"/>
      </geometry>
    </visual>
  </link>
  <joint name="index_dof3_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0.02827 0 0"/>
    <parent link="index_dof2_link"/>
    <child link="index_dof3_link"/>
    <limit effort="100" lower="-1.7453" upper="0.0" velocity="100"/>
    <axis xyz="0 -1 0"/>
  </joint>
  <link name="index_dof3_link">
    <visual>
      <origin rpy="0.0 0.0 0.0 " xyz="0 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <sphere radius="0.004" xyz="0 0 0"/>
      </geometry>
    </visual>
    <visual>
      <origin rpy="0.0 1.57079632679 0.0 " xyz="0.01306 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <cylinder length="0.02612" radius="0.002"/>
      </geometry>
    </visual>
  </link>
  <joint name="index_dof4_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="index_dof3_link"/>
    <child link="index_dof4_link"/>
    <limit effort="100" lower="-0.001" upper="0.001" velocity="100"/>
    <axis xyz="0 1 0"/>
  </joint>
  <link name="index_dof4_link">
    <visual>
      <origin rpy="0.0 0.0 0.0 " xyz="0 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <sphere radius="0.004" xyz="0 0 0"/>
      </geometry>
    </visual>
  </link>
  <joint name="index_dof5_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="index_dof4_link"/>
    <child link="index_dof5_link"/>
    <limit effort="100" lower="-0.001" upper="0.001" velocity="100"/>
    <axis xyz="0 1 0"/>
  </joint>
  <link name="index_dof5_link">
    <visual>
      <origin rpy="0.0 0.0 0.0 " xyz="0 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <sphere radius="0.004" xyz="0 0 0"/>
      </geometry>
    </visual>
  </link>
  <joint name="index_dof6_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0.02612 0 0"/>
    <parent link="index_dof5_link"/>
    <child link="index_dof6_link"/>
  </joint>
  <link name="index_dof6_link">
    <visual>
      <origin rpy="0.0 0.0 0.0 " xyz="0 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <sphere radius="0.004" xyz="0 0 0"/>
      </geometry>
    </visual>
  </link>
  <joint name="middle_dof0_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0.10125 0 0"/>
    <parent link="ahand_palm_link"/>
    <child link="middle_dof0_link"/>
    <limit effort="100" lower="-0.13963" upper="0.05236" velocity="100"/>
    <axis xyz="0 0 1"/>
  </joint>
  <link name="middle_dof0_link">
    <visual>
      <origin rpy="0.0 0.0 0.0 " xyz="0 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <sphere radius="0.004" xyz="0 0 0"/>
      </geometry>
    </visual>
  </link>
  <joint name="middle_dof1_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="middle_dof0_link"/>
    <child link="middle_dof1_link"/>
    <limit effort="100" lower="-0.13963" upper="0.05236" velocity="100"/>
    <axis xyz="0 -1 0"/>
  </joint>
  <link name="middle_dof1_link">
    <visual>
      <origin rpy="0.0 0.0 0.0 " xyz="0 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <sphere radius="0.004" xyz="0 0 0"/>
      </geometry>
    </visual>
    <visual>
      <origin rpy="0.0 1.57079632679 0.0 " xyz="0.02494 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <cylinder length="0.04988" radius="0.002"/>
      </geometry>
    </visual>
  </link>
  <joint name="middle_dof2_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0.04988 0 0"/>
    <parent link="middle_dof1_link"/>
    <child link="middle_dof2_link"/>
    <limit effort="100" lower="-1.5708" upper="0.0" velocity="100"/>
    <axis xyz="0 -1 0"/>
  </joint>
  <link name="middle_dof2_link">
    <visual>
      <origin rpy="0.0 0.0 0.0 " xyz="0 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <sphere radius="0.004" xyz="0 0 0"/>
      </geometry>
    </visual>
    <visual>
      <origin rpy="0.0 1.57079632679 0.0 " xyz="0.016295 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <cylinder length="0.03259" radius="0.002"/>
      </geometry>
    </visual>
  </link>
  <joint name="middle_dof3_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0.03259 0 0"/>
    <parent link="middle_dof2_link"/>
    <child link="middle_dof3_link"/>
    <limit effort="100" lower="-1.7453" upper="0.0" velocity="100"/>
    <axis xyz="0 -1 0"/>
  </joint>
  <link name="middle_dof3_link">
    <visual>
      <origin rpy="0.0 0.0 0.0 " xyz="0 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <sphere radius="0.004" xyz="0 0 0"/>
      </geometry>
    </visual>
    <visual>
      <origin rpy="0.0 1.57079632679 0.0 " xyz="0.015215 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <cylinder length="0.03043" radius="0.002"/>
      </geometry>
    </visual>
  </link>
  <joint name="middle_dof4_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="middle_dof3_link"/>
    <child link="middle_dof4_link"/>
    <limit effort="100" lower="-0.001" upper="0.001" velocity="100"/>
    <axis xyz="0 1 0"/>
  </joint>
  <link name="middle_dof4_link">
    <visual>
      <origin rpy="0.0 0.0 0.0 " xyz="0 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <sphere radius="0.004" xyz="0 0 0"/>
      </geometry>
    </visual>
  </link>
  <joint name="middle_dof5_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="middle_dof4_link"/>
    <child link="middle_dof5_link"/>
    <limit effort="100" lower="-0.001" upper="0.001" velocity="100"/>
    <axis xyz="0 1 0"/>
  </joint>
  <link name="middle_dof5_link">
    <visual>
      <origin rpy="0.0 0.0 0.0 " xyz="0 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <sphere radius="0.004" xyz="0 0 0"/>
      </geometry>
    </visual>
  </link>
  <joint name="middle_dof6_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0.03043 0 0"/>
    <parent link="middle_dof5_link"/>
    <child link="middle_dof6_link"/>
  </joint>
  <link name="middle_dof6_link">
    <visual>
      <origin rpy="0.0 0.0 0.0 " xyz="0 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <sphere radius="0.004" xyz="0 0 0"/>
      </geometry>
    </visual>
  </link>
  <joint name="ring_dof0_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0.093608 -0.025021  0"/>
    <parent link="ahand_palm_link"/>
    <child link="ring_dof0_link"/>
    <limit effort="100" lower="-0.13963" upper="0.05236" velocity="100"/>
    <axis xyz="0 0 1"/>
  </joint>
  <link name="ring_dof0_link">
    <visual>
      <origin rpy="0.0 0.0 0.0 " xyz="0 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <sphere radius="0.004" xyz="0 0 0"/>
      </geometry>
    </visual>
  </link>
  <joint name="ring_dof1_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="ring_dof0_link"/>
    <child link="ring_dof1_link"/>
    <limit effort="100" lower="-0.13963" upper="0.05236" velocity="100"/>
    <axis xyz="0 -1 0"/>
  </joint>
  <link name="ring_dof1_link">
    <visual>
      <origin rpy="0.0 0.0 0.0 " xyz="0 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <sphere radius="0.004" xyz="0 0 0"/>
      </geometry>
    </visual>
    <visual>
      <origin rpy="0.0 1.57079632679 0.0 " xyz="0.02208 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <cylinder length="0.04416" radius="0.002"/>
      </geometry>
    </visual>
  </link>
  <joint name="ring_dof2_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0.04416 0 0"/>
    <parent link="ring_dof1_link"/>
    <child link="ring_dof2_link"/>
    <limit effort="100" lower="-1.7453" upper="0.0" velocity="100"/>
    <axis xyz="0 -1 0"/>
  </joint>
  <link name="ring_dof2_link">
    <visual>
      <origin rpy="0.0 0.0 0.0 " xyz="0 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <sphere radius="0.004" xyz="0 0 0"/>
      </geometry>
    </visual>
    <visual>
      <origin rpy="0.0 1.57079632679 0.0 " xyz="0.01656 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <cylinder length="0.03312" radius="0.002"/>
      </geometry>
    </visual>
  </link>
  <joint name="ring_dof3_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0.03312 0 0"/>
    <parent link="ring_dof2_link"/>
    <child link="ring_dof3_link"/>
    <limit effort="100" lower="-1.7453" upper="0.0" velocity="100"/>
    <axis xyz="0 -1 0"/>
  </joint>
  <link name="ring_dof3_link">
    <visual>
      <origin rpy="0.0 0.0 0.0 " xyz="0 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <sphere radius="0.004" xyz="0 0 0"/>
      </geometry>
    </visual>
  </link>
  <joint name="ring_dof4_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="ring_dof3_link"/>
    <child link="ring_dof4_link"/>
    <limit effort="100" lower="-0.001" upper="0.001" velocity="100"/>
    <axis xyz="0 1 0"/>
  </joint>
  <link name="ring_dof4_link">
    <visual>
      <origin rpy="0.0 0.0 0.0 " xyz="0 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <sphere radius="0.004" xyz="0 0 0"/>
      </geometry>
    </visual>
  </link>
  <joint name="ring_dof5_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="ring_dof4_link"/>
    <child link="ring_dof5_link"/>
    <limit effort="100" lower="-0.001" upper="0.001" velocity="100"/>
    <axis xyz="0 1 0"/>
  </joint>
  <link name="ring_dof5_link">
    <visual>
      <origin rpy="0.0 0.0 0.0 " xyz="0 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <sphere radius="0.004" xyz="0 0 0"/>
      </geometry>
    </visual>
    <visual>
      <origin rpy="0.0 1.57079632679 0.0 " xyz="0.014435 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <cylinder length="0.02887" radius="0.002"/>
      </geometry>
    </visual>
  </link>
  <joint name="ring_dof6_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0.02887 0 0"/>
    <parent link="ring_dof5_link"/>
    <child link="ring_dof6_link"/>
  </joint>
  <link name="ring_dof6_link">
    <visual>
      <origin rpy="0.0 0.0 0.0 " xyz="0 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <sphere radius="0.004" xyz="0 0 0"/>
      </geometry>
    </visual>
  </link>
  <joint name="pinky_dof0_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0.082146 -0.047768  0"/>
    <parent link="ahand_palm_link"/>
    <child link="pinky_dof0_link"/>
    <limit effort="100" lower="-0.13963" upper="0.05236" velocity="100"/>
    <axis xyz="0 0 1"/>
  </joint>
  <link name="pinky_dof0_link">
    <visual>
      <origin rpy="0.0 0.0 0.0 " xyz="0 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <sphere radius="0.004" xyz="0 0 0"/>
      </geometry>
    </visual>
  </link>
  <joint name="pinky_dof1_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="pinky_dof0_link"/>
    <child link="pinky_dof1_link"/>
    <limit effort="100" lower="-0.13963" upper="0.05236" velocity="100"/>
    <axis xyz="0 -1 0"/>
  </joint>
  <link name="pinky_dof1_link">
    <visual>
      <origin rpy="0.0 0.0 0.0 " xyz="0 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <sphere radius="0.004" xyz="0 0 0"/>
      </geometry>
    </visual>
    <visual>
      <origin rpy="0.0 1.57079632679 0.0 " xyz="0.016945 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <cylinder length="0.03389" radius="0.002"/>
      </geometry>
    </visual>
  </link>
  <joint name="pinky_dof2_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0.03389 0 0"/>
    <parent link="pinky_dof1_link"/>
    <child link="pinky_dof2_link"/>
    <limit effort="100" lower="-1.5708" upper="0.0" velocity="100"/>
    <axis xyz="0 -1 0"/>
  </joint>
  <link name="pinky_dof2_link">
    <visual>
      <origin rpy="0.0 0.0 0.0 " xyz="0 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <sphere radius="0.004" xyz="0 0 0"/>
      </geometry>
    </visual>
    <visual>
      <origin rpy="0.0 1.57079632679 0.0 " xyz="0.01198 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <cylinder length="0.02396" radius="0.002"/>
      </geometry>
    </visual>
  </link>
  <joint name="pinky_dof3_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0.02396 0 0"/>
    <parent link="pinky_dof2_link"/>
    <child link="pinky_dof3_link"/>
    <limit effort="100" lower="-1.7453" upper="0.0" velocity="100"/>
    <axis xyz="0 -1 0"/>
  </joint>
  <link name="pinky_dof3_link">
    <visual>
      <origin rpy="0.0 0.0 0.0 " xyz="0 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <sphere radius="0.004" xyz="0 0 0"/>
      </geometry>
    </visual>
  </link>
  <joint name="pinky_dof4_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="pinky_dof3_link"/>
    <child link="pinky_dof4_link"/>
    <limit effort="100" lower="-0.001" upper="0.001" velocity="100"/>
    <axis xyz="0 1 0"/>
  </joint>
  <link name="pinky_dof4_link">
    <visual>
      <origin rpy="0.0 0.0 0.0 " xyz="0 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <sphere radius="0.004" xyz="0 0 0"/>
      </geometry>
    </visual>
  </link>
  <joint name="pinky_dof5_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="pinky_dof4_link"/>
    <child link="pinky_dof5_link"/>
    <limit effort="100" lower="-0.001" upper="0.001" velocity="100"/>
    <axis xyz="0 1 0"/>
  </joint>
  <link name="pinky_dof5_link">
    <visual>
      <origin rpy="0.0 0.0 0.0 " xyz="0 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <sphere radius="0.004" xyz="0 0 0"/>
      </geometry>
    </visual>
    <visual>
      <origin rpy="0.0 1.57079632679 0.0 " xyz="0.01271 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <cylinder length="0.02542" radius="0.002"/>
      </geometry>
    </visual>
  </link>
  <joint name="pinky_dof6_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0.02542 0 0"/>
    <parent link="pinky_dof5_link"/>
    <child link="pinky_dof6_link"/>
  </joint>
  <link name="pinky_dof6_link">
    <visual>
      <origin rpy="0.0 0.0 0.0 " xyz="0 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <sphere radius="0.004" xyz="0 0 0"/>
      </geometry>
    </visual>
  </link>
  <joint name="thumb_dof0_joint" type="revolute">
    <origin rpy="-0.0143 0 1.0471975512" xyz="0.032476 0.028433  0"/>
    <parent link="ahand_palm_link"/>
    <child link="thumb_dof0_link"/>
    <limit effort="100" lower="-1.7453" upper="0.69813" velocity="100"/>
    <axis xyz="0 0 1"/>
  </joint>
  <link name="thumb_dof0_link">
    <visual>
      <origin rpy="0.0 0.0 0.0 " xyz="0 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <sphere radius="0.004" xyz="0 0 0"/>
      </geometry>
    </visual>
  </link>
  <joint name="thumb_dof1_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="thumb_dof0_link"/>
    <child link="thumb_dof1_link"/>
    <limit effort="100" lower="-0.8203" upper="0.5236" velocity="100"/>
    <axis xyz="0 -1 0"/>
  </joint>
  <link name="thumb_dof1_link">
    <visual>
      <origin rpy="0.0 0.0 0.0 " xyz="0 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <sphere radius="0.004" xyz="0 0 0"/>
      </geometry>
    </visual>
    <visual>
      <origin rpy="0.0 1.57079632679 0.0 " xyz="0.022825 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <cylinder length="0.04565" radius="0.002"/>
      </geometry>
    </visual>
  </link>
  <joint name="thumb_dof2_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0.04565 0 0"/>
    <parent link="thumb_dof1_link"/>
    <child link="thumb_dof2_link"/>
    <limit effort="100" lower="-1.5708" upper="0.0" velocity="100"/>
    <axis xyz="0 -1 0"/>
  </joint>
  <link name="thumb_dof2_link">
    <visual>
      <origin rpy="0.0 0.0 0.0 " xyz="0 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <sphere radius="0.004" xyz="0 0 0"/>
      </geometry>
    </visual>
    <visual>
      <origin rpy="0.0 1.57079632679 0.0 " xyz="0.01339 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <cylinder length="0.02678" radius="0.002"/>
      </geometry>
    </visual>
  </link>
  <joint name="thumb_dof3_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0.02678 0 0"/>
    <parent link="thumb_dof2_link"/>
    <child link="thumb_dof3_link"/>
    <limit effort="100" lower="-1.5708" upper="0.34907" velocity="100"/>
    <axis xyz="0 -1 0"/>
  </joint>
  <link name="thumb_dof3_link">
    <visual>
      <origin rpy="0.0 0.0 0.0 " xyz="0 0 0"/>
      <material name="Grey">
        <color rgba="0.9 0.9 0.9 0.5"/>
      </material>
      <geometry>
        <sphere radius="0.004" xyz="0 0 0"/>
      </geometry>
    </visual>
  </link>
</robot>

